cognite/client/_api/synthetic_time_series.py

```diff
@@ -46,46 +45,68 @@ def retrieve(
-                >>> dps = c.datapoints.synthetic.retrieve(expression="A+B", start="2w-ago", end="now", variables=vars)
+                >>> dps = c.datapoints.synthetic.query(expressions=["A+B"], start="2w-ago", end="now", variables=vars)
 
             Use sympy to build complex expressions:
 
-                >>> from cognite.client.experimental import CogniteClient
+                >>> from cognite.client import CogniteClient
                 >>> c = CogniteClient()
                 >>> from sympy import symbols, cos, pi
                 >>> a = sympy.symbols('a')
-                >>> dps = c.datapoints.synthetic.retrieve(pi * cos(a), start="2w-ago", end="now", variables={"a": "my_ts_external_id"},aggregate='interpolation',granularity='1m')
+                >>> dps = c.datapoints.synthetic.query([pi * cos(a)], start="2w-ago", end="now", variables={"a": "my_ts_external_id"},aggregate='interpolation',granularity='1m')
             """
         if limit is None or limit == -1:
             limit = float("inf")
-        expression, short_expression = SyntheticDatapointsAPI._build_expression(
-            expression, variables, aggregate, granularity
-        )
-        query = {
-            "expression": expression,
-            "start": cognite.client.utils._time.timestamp_to_ms(start),
-            "end": cognite.client.utils._time.timestamp_to_ms(end),
-        }
-        datapoints = Datapoints(value=[], error=[])
-        datapoints.external_id = short_expression  # for dataframe readability
+
+        queries = []
+        datapoints = []
+        expr_number = len(expressions)
+        lengths = [limit] * expr_number
+        limit_per_expression = int(self._DPS_LIMIT / expr_number)
+
+        for e in expressions:
+            expression, short_expression = SyntheticDatapointsAPI._build_expression(
+                e, variables, aggregate, granularity
+            )
+            query = {
+                "expression": expression,
+                "start": cognite.client.utils._time.timestamp_to_ms(start),
+                "end": cognite.client.utils._time.timestamp_to_ms(end),
+                "limit": min(limit, limit_per_expression),
+            }
+            queries.append(query)
+            query_datapoints = Datapoints(value=[], error=[])
+            query_datapoints.external_id = short_expression  # for dataframe readability
+            datapoints.append(query_datapoints)
+
         while True:
-            query["limit"] = min(limit, self._DPS_LIMIT)
-            resp = self._post(url_path=self._SYNTHETIC_RESOURCE_PATH + "/query", json={"items": [query]})
-            data = resp.json()["items"][0]
-            datapoints._extend(Datapoints._load(data, expected_fields=["value", "error"]))
-            limit -= len(data["datapoints"])
-            if len(data["datapoints"]) < self._DPS_LIMIT or limit <= 0:
+            resp = self._post(url_path=self._RESOURCE_PATH + "/query", json={"items": queries})
```

Comment:
if you insist on implementing multi query support, I'd at least expect it to be parallelized like other endpoints.

Reply:
@sanderland , right, but how is it possible to parallelize it if we're passing multiple expressions to single request and decide if we should fetch additional datapoints based on response?

Reply:
By not passing multiple expressions to a single request. See e.g. the normal data points api.

Reply:
so, we can divide it into two endpoints – `retrieve` and `retrive_multiple`?

Reply:
No, that would be weird. Try to mirror how the normal datapoints API works.

Reply:
got it, redone with the suggested solution